src/commands/case_management.py

```diff
@@ -0,0 +1,699 @@
+    elif int(ctx.words[1]) in ctx.bot.board:
+        if rescue.client.casefold() in ctx.bot.last_user_message:
+            last_message = last_message
+        else:
+            await ctx.reply("Nothing to grab from that client.")
+            return
+    else:
+        await ctx.reply("Nothing to grab from that client.")
+        return
+
+    async with ctx.bot.board.modify_rescue(rescue) as case:
+        case.add_quote(last_message, ctx.words[1].casefold())
+        await ctx.reply(f"{case.client}'s case updated with "
+                        f"{last_message!r} (Case {case.board_index})")
+
+
+@require_channel
+@require_permission(RAT)
+@command("inject")
+async def cmd_case_management_inject(ctx: Context):
+    if len(ctx.words) < 3:
+        await ctx.reply("Usage: !inject <Client Name|Board Index> <Text to Add>")
+        return
+
+    # Pass case to validator, return a case if found or None
+    rescue = _validate(ctx, ctx.words[1])
+
+    if not rescue:
+        logger.debug("creating rescue for {!r}", ctx.words[1])
+        rescue = await ctx.bot.board.create_rescue(client=ctx.words[1])
+        async with ctx.bot.board.modify_rescue(rescue) as case:
+            case.add_quote(ctx.words_eol[2], ctx.user.nickname)
+
+            for keyword in ctx.words_eol[2].split():
+                if keyword.upper() in {item.value for item in Platforms}:
+                    case.platform = Platforms[keyword.upper()]
+                if keyword.casefold() == "cr" or "code red" in keyword.casefold() \
+                        or _TIME_RE.match(ctx.words_eol[2]):
+                    case.code_red = True
+
+            await ctx.reply(f"{case.client}'s case opened with: "
+                            f"{ctx.words_eol[2]}  (Case {case.board_index})")
+
+            if case.code_red:
+                await ctx.reply(f"Code Red! {case.client} is on Emergency Oxygen!")
+
+            return
+
+    async with ctx.bot.board.modify_rescue(rescue) as case:
+        case.add_quote(ctx.words_eol[2], ctx.user.nickname)
+
+    await ctx.reply(f"{case.client}'s case updated with: "
+                    f"'{ctx.words_eol[2]}' (Case {case.board_index})")
+
+
+@require_channel
+@require_permission(RAT)
+@command("ircnick", "nick", "nickname")
+async def cmd_case_management_ircnick(ctx: Context):
+    if len(ctx.words) < 3:
+        await ctx.reply("Usage: !ircnick <Client Name|Board Index> <New Client Name>")
+        return
+
+    # Pass case to validator, return a case if found or None
+    rescue = _validate(ctx, ctx.words[1])
+
+    if not rescue:
+        await ctx.reply("No case with that name or number.")
+        return
+
+    new_name = ctx.words[2]
+
+    async with ctx.bot.board.modify_rescue(rescue) as case:
+        case.irc_nickname = new_name
+        await ctx.reply(f"Set IRC Nickname to {case.irc_nickname!r}")
+
+    if new_name.casefold() != rescue.client.casefold():
+        await ctx.reply("Caution: IRC Nickname does not match CMDR Name.")
```

Comment:
no. generally the IRC nickname matches the Commander name, and it is thus a warning in mecha2 if these two values diverge.

Reply:
That seems like a fair warning when the case is first announced, but the entire point of this command is to change the IRC Nickname away from the CMDR name, is it not? Since the client usually joins with their CMDR Name as their IRC Name to begin with (barring any forbidden characters).